File: psyconnect-backend/src/main/java/ni/edu/uam/psyconnect_backend/repository/TestResultRepository.java
package ni.edu.uam.psyconnect_backend.repository;

import ni.edu.uam.psyconnect_backend.model.TestResult;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface TestResultRepository
        extends JpaRepository<TestResult, Long> {

    List<TestResult> findByUserIdOrderByCreatedAtDesc(
            Long userId
    );
}

File: PsyConnect/app/src/main/java/ni/edu/uam/psyconnect/data/repository/TestResultRepository.kt
package ni.edu.uam.psyconnect.data.moodjournal

import kotlinx.coroutines.flow.Flow
import ni.edu.uam.psyconnect.data.model.TestResultEntity

class TestResultRepository(
    private val dao: TestResultDao
) {
    fun getResultsByUser(userId: Long): Flow<List<TestResultEntity>> {
        return dao.getResultsByUser(userId)
    }

    fun getResultsByCategory(userId: Long, category: String): Flow<List<TestResultEntity>> {
        return dao.getResultsByCategory(userId, category)
    }

    suspend fun insert(result: TestResultEntity) {
        dao.insert(result)
    }

    suspend fun delete(result: TestResultEntity) {
        dao.delete(result)
    }
}
